crates/fuel-streams-storage/src/s3/s3_client_opts.rs

```diff
@@ -0,0 +1,71 @@
+use fuel_networks::{FuelNetwork, FuelNetworkUserRole};
+
+// Introduced for consistency.
+// TODO: make it more ergonomic by probably using FuelNetwork in S3Client directly
+#[derive(Debug, Clone, Default)]
+pub struct S3ClientOpts {
+    pub fuel_network: FuelNetwork,
+    pub role: FuelNetworkUserRole,
+    pub namespace: Option<String>,
+}
+
+impl S3ClientOpts {
+    pub fn new(fuel_network: FuelNetwork) -> Self {
+        Self {
+            fuel_network,
+            role: FuelNetworkUserRole::default(),
+            namespace: None,
+        }
+    }
+
+    pub fn admin_opts() -> Self {
+        Self::new(FuelNetwork::load_from_env())
+            .with_role(FuelNetworkUserRole::Admin)
+    }
+
+    // TODO: Consider revamping and reusing NATs' Namespace here
+    #[cfg(any(test, feature = "test-helpers"))]
+    pub fn with_random_namespace(mut self) -> Self {
+        let random_namespace = {
+            use rand::Rng;
+            let random_int: u32 = rand::thread_rng().gen();
+            format!("namespace-{}", random_int)
+        };
+        self.namespace = Some(random_namespace);
+        self
+    }
+
+    pub fn with_role(self, role: FuelNetworkUserRole) -> Self {
+        Self { role, ..self }
+    }
+
+    pub fn endpoint_url(&self) -> String {
+        match self.role {
```

Comment:
This should be optional - if it's not provided, don't set it. We should never set this manually when we're deploying, it's only useful for tests or when running outside of AWS.

Reply:
Right. This code also runs outside AWS for client apps. But I'll make this the default `AWS_ENDPOINT_URL` and optionally set it as you've recommended.